src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,105 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final double GROW_MULTIPLY = 1.5;
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final String ERROR_ELEMENT_NOT_EXIST = "Element does not exist";
+    private int size;
+    private T[] array;
+
+    public ArrayList() {
+        array = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        growCheck();
+        array[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        growCheck();
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Element not found at " + index);
+        }
+        System.arraycopy(array, index, array, index + 1, size - index);
+        array[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (isNotValidValue(index)) {
+            throw new ArrayListIndexOutOfBoundsException(ERROR_ELEMENT_NOT_EXIST);
+        }
```

Comment:
This check is very similar to the next ones, let's create a separate method for it